elastalert/elastalert.py

```diff
@@ -414,7 +432,7 @@ def set_starttime(self, rule, endtime):
             if 'minimum_starttime' in rule and rule['minimum_starttime'] > buffer_delta:
                 rule['starttime'] = rule['minimum_starttime']
             # If buffer_time doesn't bring us past the previous endtime, use that instead
-            elif 'previous_endtime' in rule and rule['previous_endtime'] < buffer_delta:
+            elif 'previous_endtime' in rule and rule['previous_endtime'] > buffer_delta:
```

Comment:
I have not changed it and just merged it with the master of Yelp/elastalert. Please let me know if i have to revert this.